crates/viewer/re_time_panel/benches/bench_density_graph.rs

```diff
@@ -0,0 +1,257 @@
+#![allow(clippy::unwrap_used)] // acceptable for code which is not user-facing
+
+use std::hint::black_box;
+use std::sync::Arc;
+use std::time::Duration;
+
+use criterion::measurement::WallTime;
+use criterion::Bencher;
+use criterion::Criterion;
+use re_chunk_store::ChunkStoreConfig;
+use re_entity_db::EntityDb;
+use re_log_types::ResolvedTimeRange;
+use re_log_types::StoreId;
+use re_log_types::StoreKind;
+use re_log_types::Timeline;
+use re_time_panel::__bench::{
+    build_density_graph, DensityGraphBuilderConfig, TimePanelItem, TimeRangesUi,
+};
+use re_viewer_context::TimeView;
+
+fn run(
+    b: &mut Bencher<'_, WallTime>,
+    config: DensityGraphBuilderConfig,
+    data_entries: &[ChunkEntry],
+) {
+    egui::__run_test_ui(|ui| {
+        let row_rect = ui.max_rect();
+        assert!(row_rect.width() > 100.0 && row_rect.height() > 100.0);
+
+        let mut db = EntityDb::with_store_config(
+            StoreId::from_string(StoreKind::Recording, "test".into()),
+            ChunkStoreConfig::COMPACTION_DISABLED,
+        );
+        let entity_path = re_log_types::EntityPath::parse_strict("/data").unwrap();
+
+        for ChunkEntry {
+            num_chunks,
+            num_rows_per_chunk,
+            sorted,
+            time_start_ms,
+        } in data_entries.iter().copied()
+        {
+            add_data(
+                &mut db,
+                &entity_path,
+                num_chunks,
+                num_rows_per_chunk,
+                sorted,
+                time_start_ms,
+            )
+            .unwrap();
+        }
+
+        let item = TimePanelItem {
+            entity_path,
+            component_name: None,
+        };
+
+        let times = db.times_per_timeline().get(&Timeline::log_time()).unwrap();
+        let time_range = ResolvedTimeRange::new(
+            times
+                .keys()
+                .next()
+                .copied()
+                .unwrap_or(re_log_types::TimeInt::MIN),
```

Comment:
```suggestion
                .unwrap(),
```

or we have a bug in the setup